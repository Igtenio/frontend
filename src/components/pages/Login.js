import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { Button, Container, Form } from 'react-bootstrap';

import { actions as appActions } from 'reducers/application';

export class Login extends Component {
  static propTypes = {
    actions: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleTextChange = this.handleTextChange.bind(this);
    this.state = {
      emailAddress: '',
      password: ''
    };
  }

  handleSubmit(e) {
    const { emailAddress, password } = this.state;
    const { actions } = this.props;

    actions.loginUser(emailAddress, password);
    e.preventDefault();
  }

  handleTextChange(e) {
    const { target } = e;
    const { name, value } = target;

    this.setState({ [name]: value });
  }

  render() {
    const { emailAddress, password } = this.state;

    return (
      <Container>
        <h1>Login</h1>
        <Form>
          <Form.Group>
            <Form.Label>Email address</Form.Label>
            <Form.Control
              type="email"
              name="emailAddress"
              placeholder="someone@widget.co"
              value={emailAddress}
              onChange={this.handleTextChange}
            />
            <Form.Text className="text-muted">
              We&apos;ll never share your email with anyone else.
            </Form.Text>
          </Form.Group>
          <Form.Group>
            <Form.Label>Password</Form.Label>
            <Form.Control
              type="password"
              placeholder="Your password"
              name="password"
              value={password}
              onChange={this.handleTextChange}
            />
          </Form.Group>
          <Button variant="primary" type="submit" onClick={this.handleSubmit}>
            Login
          </Button>
        </Form>
      </Container>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      ...appActions
    },
    dispatch
  )
});

export default connect(
  null,
  mapDispatchToProps
)(Login);
