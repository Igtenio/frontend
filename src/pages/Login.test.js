import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Login from './Login';
import { withMemoryRouter } from 'utils';

describe('<Login />', () => {
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('renders correctly', () => {
    const RoutedLogin = withMemoryRouter(Login);
    const component = renderer.create(
      <Provider store={store}>
        <RoutedLogin />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
