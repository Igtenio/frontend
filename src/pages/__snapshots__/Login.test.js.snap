// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<div
  className="container"
>
  <h1>
    Login
  </h1>
  <form
    className=""
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Email address
      </label>
      <input
        className="form-control"
        name="emailAddress"
        onChange={[Function]}
        placeholder="someone@widget.co"
        type="email"
        value=""
      />
      <small
        className="text-muted form-text"
      >
        We'll never share your email with anyone else.
      </small>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        placeholder="Your password"
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
